### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by naterush**
This PR does two things:
1. It cleans up the React code we have. I made things into components that should be components, refactored some constants out, and make screens where there should be screens (although I don't think that "screen" is the right word). 
2. It hides sheets when they are saga sheets. Right now, I'm not setting them to "veryHidden" but rather just "hidden" so someone could unhide them through the Excel UI if they wanted to - but deleting veryHidden sheets is harder and I wanted to get this features out the door ASAP -- to address user issue!

### CODE REVIEW

File: src/routers/project.js [hunk 1]
```diff
@@ -5,10 +5,12 @@ const Projects = mongoose.model('Projects');
...
 
-const BRANCH_STATE_HEAD = 0;
-const BRANCH_STATE_AHEAD = 1;
-const BRANCH_STATE_BEHIND = 2;
-const BRANCH_STATE_FORKED = 3;
+// TODO: import these from constants!
+const BRANCH_STATE_HEAD = 'branch_state_head';
+const BRANCH_STATE_AHEAD = 'branch_state_ahead';
+const BRANCH_STATE_BEHIND ='branch_state_behind';
+const BRANCH_STATE_FORKED ='branch_state_forked';
```
Comment (naterush) [inline src/routers/project.js:12]: I unfortunately redefine constants here b/c I'm not sure how the new JS we use with webpack interacts with old JS for node. Will figure out moving forward. Marked in issue: https://github.com/saga-vcs/saga-excel/issues/11

**Comment by naterush** [comment]
Scenarios to check:
1. Creating a new project actually pushes (e.g. can download from another Excel window with link). 
2. Merge works w/ no merge conflicts.
3. Merge works with merge conflicts. 

That's enough, for now, I think.

**Comment by aarondr77** [comment]
The updates you made look good, but testing found bugs. 

**Bug Report 1:** 
Steps to reproduce:
1. Create a new project
2. Enter data in column A1
3. Merge
4. Try unsuccessfully to switch versions or reset personal version-> "Invalid Operation: This operation is not permitted for the current object"

Let's do another round of testing once this bug is squashed because it is a blocker for more in depth testing. 

**Bug Report 2:** 
Steps to reproduce:
1. Create a new project
2. Enter 1 in A1
3. Merge
4. Unhide Saga Sheet and find master sheet
5. Unhide Master sheet
6. On master sheet, add 2 in B1
7. On Personal sheet, add 3 in B1
8. Merge

The merge is unsuccessful. The taskpane shows the merge error screen. Most troublingly, the saga worksheet is gone. It is not displayed or a hidden sheet. From then on, it throws the "The Requested Resource Does not Exist" error on every sync. 

This is a bug with deleteNonSagaSheets. If I run the same procedure but at step 7.5, I hide the saga sheet, then the merge is successful and takes B1 to be 3, the value of the personalBranch. Is this the intended behavior? 

Let's do another round of testing once this bug is squashed because it is a blocker for more in depth testing.

**Comment by naterush** [comment]
@aarondr77 Got it. Checking them out now. Good catches.

**Comment by naterush** [comment]
@aarondr77 both bugs should be fixed. Both of them came from the same general area: you can't delete a sheet if it's the only non-hidden sheet. 

To fix this (e.g. we delete all visible sheets during checkout), I was renaming the active sheet from, say, Sheet1 to saga-tmp, then copying over the new Sheet1 from the newly checked out branch, and then deleting saga-tmp. In practice, I wasn't getting Sheet1 by name, and it turns out I was screwing up, renaming saga to saga-tmp, and then deleting it. Silly!

**Pull request merged**